dockerfiles/alpine_3.9

```diff
@@ -132,9 +132,7 @@ RUN set -x \
     && rm -r /usr/src/luarocks \
     && rm -rf /usr/src/tarantool \
     && rm -rf /usr/src/gperftools \
-    && rm -rf /usr/src/go \
-    && : "---------- remove build deps ----------" \
-    && apk del .build-deps
```

Comment:
Virtual packages are used here in order to exclude build time dependencies from the resulting image and keep the image smaller. Let's look whether we can overcome the problem, but keep the approach.

From the link you kindly included into the description:

> The problem is likely that the virtual package checksum (basically a made up hash to uniquely identify the package) is built from the second precision timestamp. So the ID clashes when two virtual packages get added within same second.

So `sleep 2` should resolve the problem. Cons: a bit longer build time. Pros: smaller resulting image.

We can also wrap it into an Alpine version check, e. g. `case "$(cat /etc/alpine-release)" in 3.9*|3.10*) sleep 2 ;; esac` (without any bashishms!), so it'll not require any further maintainance.

Reply:
I don't think workaround with sleep can help us with the the issue. Please check the following checks I've made on the image:
1) Failed to install packages with `--virtual` option and with `sleep 2`
```
Alexander Tikhonov, [21.10.20 09:43]
а вот и выдача как я проверял, что виртуал реально не работает
/opt/tarantool # for p in mariadb-client libpq ; do apk add --no-cache --virtual .run-deps $p ; sleep 2 ; done
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/community/x86_64/APKINDEX.tar.gz
OK: 74 MiB in 44 packages
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/community/x86_64/APKINDEX.tar.gz
OK: 74 MiB in 44 packages
/opt/tarantool # tarantool
Tarantool 2.6.0-0-g47aa4e01e
type 'help' for interactive help
tarantool> require('pg')
---
- error: "/.rocks/share/tarantool/pg/init.lua:4: module 'pg.driver' not found:\n\tno
    field package.preload['pg.driver']\n\tno file '/opt/tarantool/pg/driver.lua'\n\tno
    file '/opt/tarantool/pg/driver/init.lua'\n\tno file '/opt/tarantool/pg/driver.so'\n\tno
    file '/opt/tarantool/.rocks/share/tarantool/pg/driver.lua'\n\tno file '/opt/tarantool/.rocks/share/tarantool/pg/driver/init.lua'\n\tno
    file '/opt/.rocks/share/tarantool/pg/driver.lua'\n\tno file '/opt/.rocks/share/tarantool/pg/driver/init.lua'\n\tno
    file '/.rocks/share/tarantool/pg/driver.lua'\n\tno file '/.rocks/share/tarantool/pg/driver/init.lua'Error
    loading shared library libpq.so.5: No such file or directory (needed by /.rocks/lib/tarantool/pg/driver.so)\n\tno
    file '/root/.luarocks/share/lua/5.1/pg/driver.lua'\n\tno file '/root/.luarocks/share/lua/5.1/pg/driver/init.lua'\n\tno
    file '/root/.luarocks/share/lua/pg/driver.lua'\n\tno file '/root/.luarocks/share/lua/pg/driver/init.lua'\n\tno
    file '/usr/local/share/tarantool/pg/driver.lua'\n\tno file '/usr/local/share/tarantool/pg/driver/init.lua'\n\tno
    file '/usr/share/tarantool/pg/driver.lua'\n\tno file '/usr/share/tarantool/pg/driver/init.lua'\n\tno
    file '/usr/local/share/lua/5.1/pg/driver.lua'\n\tno file '/usr/local/share/lua/5.1/pg/driver/init.lua'\n\tno
    file '/usr/share/lua/5.1/pg/driver.lua'\n\tno file '/usr/share/lua/5.1/pg/driver/init.lua'\n\tno
    file '/root/.luarocks/lib/lua/5.1/pg/driver.so'\n\tno file '/root/.luarocks/lib/lua/pg/driver.so'\n\tno
    file '/usr/local/lib64/tarantool/pg/driver.so'\n\tno file '/usr/local/lib/tarantool/pg/driver.so'\n\tno
    file '/usr/lib64/tarantool/pg/driver.so'\n\tno file '/usr/local/lib64/lua/5.1/pg/driver.so'\n\tno
    file '/usr/local/lib/lua/5.1/pg/driver.so'\n\tno file '/usr/lib64/lua/5.1/pg/driver.so'\n\tno
    file '/root/.luarocks/lib/lua/5.1/pg.so'\n\tno file '/root/.luarocks/lib/lua/pg.so'\n\tno
    file '/usr/local/lib64/tarantool/pg.so'\n\tno file '/usr/local/lib/tarantool/pg.so'\n\tno
    file '/usr/lib64/tarantool/pg.so'\n\tno file '/usr/local/lib64/lua/5.1/pg.so'\n\tno
    file '/usr/local/lib/lua/5.1/pg.so'\n\tno file '/usr/lib64/lua/5.1/pg.so'"
...
```
2) Successfully installed packages without --virtual option:
```
tarantool> /opt/tarantool # for p in mariadb-client libpq ; do apk add --no-cache .run-deps $p ; sleep 2 ; done
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/community/x86_64/APKINDEX.tar.gz
(1/2) Installing mariadb-common (10.3.25-r0)
(2/2) Installing mariadb-client (10.3.25-r0)
Executing busybox-1.29.3-r10.trigger
OK: 104 MiB in 46 packages
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/community/x86_64/APKINDEX.tar.gz
(1/4) Installing db (5.3.28-r1)
(2/4) Installing libsasl (2.1.27-r2)
(3/4) Installing libldap (2.4.48-r1)
(4/4) Installing libpq (11.9-r0)
OK: 106 MiB in 50 packages
/opt/tarantool # tarantool
Tarantool 2.6.0-0-g47aa4e01e
type 'help' for interactive help
tarantool> require('pg')
---
- pool_create: 'function: 0x414d7a98'
  connect: 'function: 0x414d90f0'
...
```

Reply:
Finally the issue resolved with workaround discussed and found at:
https://stackoverflow.com/questions/46221063/what-is-build-deps-for-apk-add-virtual-command/46222036#46222036

Reply:
Found https://gitlab.alpinelinux.org/alpine/apk-tools/-/issues/9651. I'll paste more thorough analysis soon.